crates/driver/src/domain/competition/solution/encoding.rs

```diff
@@ -341,41 +297,70 @@ pub fn tx(
     })
 }
 
-/// Makes sure that there is an order that intended to pay for a
-/// given flashloan and that the payment is sufficient.
-fn ensure_flashloan_is_paid_for(
-    auction: &competition::Auction,
-    solution: &super::Solution,
-    uid: &order::Uid,
-    settlement: H160,
-    expected_payment: eth::Asset,
-) -> Result<(), String> {
-    let Some(trade) = solution.trades().iter().find(|t| t.uid() == *uid) else {
-        return Err("no trade execution to repay the flashloan".into());
-    };
-    if trade.receiver().0 != settlement {
-        return Err("order does not pay the settlement contract".into());
+fn pay_back_via_tracker(
+    flashloan: &Flashloan,
+    flashloan_wrapper: &FlashloanWrapperData,
+    flashloan_tracker: &Contract,
+) -> Interaction {
+    let pay_back_call_data = flashloan_tracker
+        .pay_back(
+            flashloan_wrapper.helper_contract.address(),
+            flashloan.borrower.into(),
+            flashloan.token.into(),
+        )
+        .tx
+        .data
+        .unwrap();
+    eth::Interaction {
+        target: flashloan_tracker.address().into(),
+        value: 0.into(),
+        call_data: Bytes(pay_back_call_data.0),
     }
+}
 
-    let paid = trade.buy();
-    if paid.token != expected_payment.token {
-        return Err("order pays with the wrong token".into());
-    }
-    if paid.amount < expected_payment.amount {
-        return Err("order does not pay enough".into());
+fn allow_lender_pull_funds(
+    contracts: &Contracts,
+    flashloan: &Flashloan,
+    flashloan_wrapper: &FlashloanWrapperData,
+    repayment_amount: U256,
+) -> Interaction {
+    let allow_aave_funds_pull = flashloan_wrapper
+        .helper_contract
+        .approve(
+            contracts.weth().address(),
```

Comment:
This one shouldn't be hardcoded weth, I think it's a merge/rebase thing that it got nuked from your fixed code.